scripts/dts/edtlib.py

```diff
@@ -1173,6 +1354,142 @@ def _init_regs(self):
 
         _add_names(node, "reg", self.regs)
 
+    def _init_partitions(self):
+        # Initializes self.partitions
+
+        self.partitions = []
+
+        if not 'partitions' in self.children:
+            # This node does not have partitions
+            return
+
+        if self.name == "partitions" or \
+              (self.parent and self.parent.name == "partitions"):
+            # This node is a partitions node or the child of a
+            # partitions node - but we expect a flash node
+            return
+
+        self.partitions = _partitions_list(self.children['partitions'], self)
+
+    def _init_pinctrl_state(self, pinctrl_node, pinctrl_state_node):
```

Comment:
Ditto here. See if it can be split up a bit.

Reply:
Split. Did not reach 30 lines.